framework/include/cppmicroservices/SharedLibrary.h

```diff
@@ -209,7 +209,7 @@ class US_Framework_EXPORT SharedLibrary
   bool IsLoaded() const;
 
 private:
-  ExplicitlySharedDataPointer<SharedLibraryPrivate> d;
+  std::shared_ptr<SharedLibraryPrivate> d;
```

Comment:
I think we should make this more straight-forward and use a std::unique_ptr here. Then always deep-copy the private impl in the SharedLibrary copy constructor.